python/hail/expr/functions.py

```diff
@@ -1896,9 +1896,18 @@ def is_transversion(ref, alt) -> BooleanExpression:
     -------
     :class:`.BooleanExpression`
     """
-    return _func("is_transversion", tbool, ref, alt)
+    return is_snp(ref, alt) & (~(_is_snp_transition(ref, alt)))
 
 
+@typecheck(ref=expr_str, alt=expr_str)
+def _is_snp_transition(ref, alt) -> BooleanExpression:
+    indices = hl.range(0, ref.length())
+    return hl.any(lambda i: hl.cond(ref[i] == alt[i], hl.literal(False),
```

Comment:
I think it's slightly neater to do something like `ref[i] != alt[i] & ...`. If you choose to leave it like this, the `hl.literal` around the `False` shouldn't be necessary, I think